### ISSUE TITLE
Running Kong with external PostgreSQL fails

### ISSUE THREAD

**Issue by svenmueller** [Kong/kong#3211]
### Summary

I'm trying to run Kong in k8s (version `1.8`, stable Helm chart https://github.com/kubernetes/charts/tree/master/stable/kong) with an external PostegreSQL database (Azure managed PostgreSQL service). The pod (container) fails to start with the following logs:

```
$ kubectl logs kong-kong-74ccf58f97-6ldtf -n ms
prefix directory /usr/local/kong not found, trying to create it
2018/02/10 20:45:04 [error] 1#0: init_by_lua error: /usr/local/share/lua/5.1/kong/init.lua:159: bad argument #2 to 'assert' (string expected, got table)
stack traceback:
	[C]: in function 'assert'
	/usr/local/share/lua/5.1/kong/init.lua:159: in function 'init'
	init_by_lua:3: in main chunk
nginx: [error] init_by_lua error: /usr/local/share/lua/5.1/kong/init.lua:159: bad argument #2 to 'assert' (string expected, got table)
stack traceback:
	[C]: in function 'assert'
	/usr/local/share/lua/5.1/kong/init.lua:159: in function 'init'
	init_by_lua:3: in main chunk
```

Used environment variables for the Kong container:

```
    Environment:
      KONG_ADMIN_LISTEN:      0.0.0.0:8001
      KONG_ADMIN_LISTEN_SSL:  0.0.0.0:8444
      KONG_PROXY_LISTEN:      0.0.0.0:8000
      KONG_PROXY_LISTEN_SSL:  0.0.0.0:8443
      KONG_NGINX_DAEMON:      off
      KONG_PROXY_ACCESS_LOG:  /dev/stdout
      KONG_ADMIN_ACCESS_LOG:  /dev/stdout
      KONG_PROXY_ERROR_LOG:   /dev/stderr
      KONG_ADMIN_ERROR_LOG:   /dev/stderr
      KONG_DATABASE:          postgres
      KONG_PG_DATABASE:       kong
      KONG_PG_HOST:           xxx
      KONG_PG_PASSWORD:       xxx
      KONG_PG_USER:           xxx
```

Is this an issue with the config i'm using? Any idea how i can continue from here or how to fix it? The logged error isn't useful to me at all.

### Additional Details & Logs

- Docker image: `kong:0.12.1`

Thx,
Sven

**Comment by shashiranjan84** [comment]
Can you paste the output of `helm install --dry-run --debug`? Redact info which don't want to share.

**Comment by svenmueller** [comment]
I found the issue by trial end error. The database credentials for the AZURE PostgresSQL database have been wrong and the init return an error message (string) instead of the expected table.

@shashiranjan84 any idea how enable SSL connection verification to the postgres database using the stable helm chart config? It seems to be only possible to enable SSL connection (`pg_ssl`) but there is no way to pass the certifcate for verifiying SSL connections. Is that correct?

Thx a lot!

**Comment by thibaultcha** [comment]
@svenmueller That error superseding the actual error message because given as a table instead of a string will be fixed in the next release. Glad you figured it out!

**Comment by shashiranjan84** [comment]
@svenmueller Though Kong does have config `lua_ssl_trusted_certificate` to add you trusted certs but  current Helm chart has no support for that. We will try to add it in next release.

For now you can download the charts and modify the Kong template to mount your certs.

**Comment by thibaultcha** [comment]
This has been fixed by now in 0.12.2 and above. Thank you for the report.
